src/change/models/changeManager.ts

```diff
@@ -0,0 +1,132 @@
+import { inject, injectable } from 'tsyringe';
+import {
+  Actions,
+  FlattenedGeoJSONPoint,
+  FlattenedGeoJSONLine,
+  FlattenedGeoJSONPolygon,
+  getChangeFromPoint,
+  getChangeFromLine,
+  getChangeFromPolygon,
+} from '@map-colonies/osm-change-generator';
+import { parseOsmWayApi, BaseElement, OsmNode, OsmWay, OsmChange, OsmElementType, OsmApiWay } from '@map-colonies/node-osm-elements';
+
+import { Services } from '../../common/constants';
+import { ILogger } from '../../common/interfaces';
+import { validateArrayHasElements } from '../../common/util';
+import { ChangeModel } from './change';
+import { OsmApiElements } from './helpers';
+import { FlattenedGeoJSON } from './geojsonTypes';
+import { ParseOsmElementsError } from './errors';
+import { isNode, isWay } from './helpers';
+
+export const generateOsmChange = (action: Actions, feature: FlattenedGeoJSON, osmElements: OsmApiElements): OsmChange => {
+  switch (feature.geometry.type) {
+    case 'Point': {
+      feature = feature as FlattenedGeoJSONPoint;
+      switch (action) {
+        case Actions.CREATE: {
+          return getChangeFromPoint({ action, feature });
+        }
+        case Actions.MODIFY: {
+          const oldElement = getNodeFromElements(osmElements);
+          return getChangeFromPoint({ action, feature, oldElement });
+        }
+        case Actions.DELETE: {
+          const oldElement = getNodeFromElements(osmElements);
+          return getChangeFromPoint({ action, oldElement });
+        }
+      }
+      break;
+    }
+    case 'LineString': {
+      feature = feature as FlattenedGeoJSONLine;
+      switch (action) {
+        case Actions.CREATE: {
+          return getChangeFromLine({ action, feature });
+        }
+        case Actions.MODIFY: {
+          const oldElement = getOsmWayFromElements(osmElements);
+          return getChangeFromLine({ action, feature, oldElement });
+        }
+        case Actions.DELETE: {
+          const oldElement = getOsmWayFromElements(osmElements);
+          return getChangeFromLine({ action, oldElement });
+        }
+      }
+      break;
+    }
+    case 'Polygon': {
+      feature = feature as FlattenedGeoJSONPolygon;
+      switch (action) {
+        case Actions.CREATE: {
+          return getChangeFromPolygon({ action, feature });
+        }
+        case Actions.MODIFY: {
+          const oldElement = getOsmWayFromElements(osmElements);
+          return getChangeFromPolygon({ action, feature, oldElement });
+        }
+        case Actions.DELETE: {
+          const oldElement = getOsmWayFromElements(osmElements);
+          return getChangeFromPolygon({ action, oldElement });
+        }
+      }
+    }
+  }
+};
+
+export const getNodeFromElements = (elements: OsmApiElements): OsmNode => {
+  if (!validateArrayHasElements(elements)) {
+    throwParseOsmElementsError('node');
```

Comment:
better add a return for denying problem with test coverage, wdyt?

Reply:
done.